### PR TITLE
splices and this assignments in conditional bindings

### PR DESCRIPTION
**Pull request by STRd6**

### CODE REVIEW

File: source/parser.hera [hunk 1]
```diff
@@ -4069,7 +4072,18 @@ ImplicitExportSpecifier
 Declaration
   HoistableDeclaration
   ClassDeclaration
-  LexicalDeclaration
+  LexicalDeclaration:d ->
+    if (d.thisAssignments?.length)
+      return {
+        ...d,
+        children: [...d.children, ...d.splices, ";", ...d.thisAssignments]
```
Comment (edemaine) [inline source/parser.hera:4079]: Should we just have a general mechanism for postfixing statements here? Or is there a need to have splices and thisAssignments separated?

Comment (edemaine) [inline source/parser.hera:4079]: Will this throw if splices is undefined? Maybe worth building up an array and protecting splices and thisAssignments pushing with ifs.

File: source/parser.hera [hunk 2]
```diff
@@ -4121,11 +4139,15 @@ LexicalBinding
...
 
+    const [splices, thisAssignments] = gatherBindingCode(binding.children)
+
     return {
       children: [binding, initializer],
       names: binding.names,
       binding,
       initializer,
+      splices: splices.map(s => [",", s]),
+      thisAssignments: thisAssignments.map(s => ["", s, ";"]),
```
Comment (edemaine) [inline source/parser.hera:4150]: Ah, maybe these different separators are a good reason to keep them separate.

**Review by edemaine — APPROVED** [review]

File: source/parser.hera (hunk 1 above)
Comment (STRd6) [reply source/parser.hera:4079]: The splices become part of the declaration so I don't think it can be generalized.

**Pull request merged**